sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -577,6 +581,27 @@ def fit(self, X, y, sample_weight=None):
             self._random_seed = rng.randint(np.iinfo(np.uint32).max, dtype="u8")
             feature_subsample_seed = rng.randint(np.iinfo(np.uint32).max, dtype="u8")
             self._feature_subsample_rng = np.random.default_rng(feature_subsample_seed)
+            # TODO: Remove this condition, once numpy 1.25 is the minimum version.
+            if parse_version(np.__version__) >= parse_version("1.25"):
+                self._bagging_subsample_rng = self._feature_subsample_rng.spawn(1)[0]
+            else:
+                # See numpy Generator.spawn(self, int n_children) and
+                # numpy BitGenerator.spawn
+
+                def spawnGenerator(self, n_children):
```

Comment:
Why the `camelCase` notation? I would have expected the more pythonic `spawn_generator` and `spawn_bit_generator` instead.

Reply:
Also I find the use of `self` confusing in this context (as it shadows the `self` variable of the surrounding `fit` method and makes the code more difficult to reason about. I would rather rename that variable to `rng` or `random_state` for instance.